src/components/Dashboard/FilingHistoryList/HeaderActions.vue

```diff
@@ -114,33 +114,77 @@ export default class HeaderActions extends Mixins(AllowableActionsMixin) {
 
   /**
    * Whether to disable correction for THIS filing.
-   * This is function instead of a getter so that we always query the realtime FF.
+   * (This is function instead of a getter so that we always query the realtime FF.)
    */
   disableCorrection (): boolean {
-    // first check allowable actions
+    // disable if not allowed
     if (!this.isAllowed(AllowableActions.CORRECTION)) return true
 
-    const conditions: Array<() => boolean> = []
-    // list of conditions to DISABLE correction
-    // (any condition not listed below is ALLOWED)
-    conditions[0] = () => this.filing.availableOnPaperOnly
-    conditions[1] = () => this.isTypeStaff
-    conditions[2] = () => (
+    // disable if filing is paper-only
+    if (this.filing.availableOnPaperOnly) return true
+
+    // disable if filing is future effective but is not completed or corrected
+    if (
       this.filing.isFutureEffective &&
       !EnumUtilities.isStatusCompleted(this.filing) &&
       !EnumUtilities.isStatusCorrected(this.filing)
-    )
-    conditions[3] = () => (EnumUtilities.isTypeIncorporationApplication(this.filing) &&
-      !this.isBaseCompany && !this.isCoop)
-    conditions[4] = () => (EnumUtilities.isTypeChangeOfRegistration(this.filing) && !this.isFirm)
-    conditions[5] = () => (EnumUtilities.isTypeCorrection(this.filing) && !this.isFirm &&
-      !this.isBaseCompany && !this.isCoop)
-    conditions[6] = () => (EnumUtilities.isTypeRegistration(this.filing) && !this.isFirm)
-    conditions[7] = () => (EnumUtilities.isTypeAmalgamation(this.filing) && !this.isBaseCompany)
-    conditions[8] = () => (EnumUtilities.isTypeContinuationIn(this.filing) && !this.isBaseCompany)
-
-    // check if any condition is True
-    return conditions.some(condition => condition())
+    ) return true
+
+    // disable/allow according to filing type
+    switch (true) {
```

Comment:
IMHO, this syntax is only desirable in very few situations. I think this is a good use of it -- it saves a  bunch of `if then else-if ...` statements.

Below is an entry for all filing types that we know about (so far), so there's no more guessing whether a filing can be corrected or not.

If we add a filing type but don't update this method, line 185 should catch it, and we have a safe fallback on line 187.